calculate_number_of_profiled_coding_base_pairs/add_number_of_profiled_coding_base_pairs.py

```diff
@@ -0,0 +1,94 @@
+import csv
+import pandas as pd
+import os
+import sys
+import argparse
+import ntpath
+
+def generate_dict(key, value, dictionary):
+    if not pd.isnull(key):
+        if ';' not in key:
+            dictionary[key] = value
+        else:
+            genes = key.split('; ')
+            for gene in genes:
+                dictionary[gene] = value
+
+def update_file(args):
+    inputFile = args.inputFile
+    uniprotFile = args.uniprotFile or os.getcwd() + "/uniprot_gene_name_with_protein_length.tab"
+    outputFile = args.outputFile or os.path.dirname(inputFile) + "/output_" + ntpath.basename(inputFile)
+
+    # get protein length by gene name
+    df_uniprot_gene_name_with_protein_length = pd.read_csv(uniprotFile, sep='\t')
+    gene_name_to_length_dict = dict()
+    df_uniprot_gene_name_with_protein_length.apply(lambda row: generate_dict(row['Gene names  (primary )'], row['Length'], gene_name_to_length_dict), axis=1)
+
+    # write files
+    input = open(inputFile, 'rt')
+    output = open(outputFile, 'wt')
+
+    for row in input.readlines():
+        output.write(row)
+        # find the gene list
+        splitRow = row.split(":")
+        if len(splitRow) == 2 and splitRow[0] == "gene_list":
+            genes = splitRow[1].split('\t')
```

Comment:
The first space was not stripped. Therefore the first gene was not recognized. For example, if the gene list field in panel file is gene_list:(space)TP53(tab)IDH2(tab)DNMT3A.... The first gene TP53 wouldn't get parsed because the space in front was identified as part of the symbol.

Reply:
@yichaoS Good point, fixed

Reply:
Thanks!